azureLogIngestion/adapter/index.ts

```diff
@@ -49,11 +50,20 @@ export default class Adapter {
      * There may be situations where a message corresponds to more than one
      * type of telemetry. In this case, the switch/case may not make sense.
      */
-    processMessages(messages: string, context: Context): void {
-        let records: Records
-
+    processMessages(messages: string | string[], context: Context): void {
         try {
-            records = JSON.parse(messages)
+            const messageArray = _.isArray(messages) ? messages : [messages]
+
+            messageArray.forEach((message) => {
```

Comment:
Beware of lexical scoping issues here and in the nested forEach below....

Reply:
While these arrow functions aren't changing scope, it's a good idea to pass scope explicitly, so I've updated that.